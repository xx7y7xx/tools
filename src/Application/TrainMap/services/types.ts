export interface TrainPosition {
  id: string;
  trainNumber: string;
  pocsag1234002Data: {
    pocsagMsgTimestamp: string;
    wgs84_latitude?: number;
    wgs84_longitude?: number;
    gcj02_latitude?: number;
    gcj02_longitude?: number;
    rawMessage?: string;
  };
}

export interface WebSocketMessage {
  type: 'train_position' | 'train_positions' | 'error' | 'ping' | 'pong';
  data?: any;
  timestamp: string;
}

export interface TrainPositionMessage extends WebSocketMessage {
  type: 'train_position';
  data: TrainPosition;
}

export interface TrainPositionsMessage extends WebSocketMessage {
  type: 'train_positions';
  data: TrainPosition[];
}
