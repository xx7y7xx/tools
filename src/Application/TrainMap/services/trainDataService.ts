import { TrainPosition, WebSocketMessage } from './types';

export class TrainDataService {
  private ws: WebSocket | null = null;
  private reconnectAttempts = 0;
  private maxReconnectAttempts = 5;
  private reconnectDelay = 1000;
  private updateCallbacks: ((train: TrainPosition) => void)[] = [];
  private allTrainsCallbacks: ((trains: TrainPosition[]) => void)[] = [];
  private connectionCallbacks: ((connected: boolean) => void)[] = [];
  private isConnected = false;

  constructor(private serverUrl: string = 'ws://localhost:8080/trains') {}

  connect(): Promise<void> {
    return new Promise((resolve, reject) => {
      try {
        this.ws = new WebSocket(this.serverUrl);

        this.ws.onopen = () => {
          console.log('🔌 Connected to Train WebSocket Server');
          this.isConnected = true;
          this.reconnectAttempts = 0;
          this.notifyConnectionStatus(true);
          resolve();
        };

        this.ws.onmessage = (event) => {
          try {
            const message: WebSocketMessage = JSON.parse(event.data);
            this.handleMessage(message);
          } catch (error) {
            console.error('❌ Error parsing WebSocket message:', error);
          }
        };

        this.ws.onclose = (event) => {
          console.log(
            '🔌 WebSocket connection closed:',
            event.code,
            event.reason
          );
          this.isConnected = false;
          this.notifyConnectionStatus(false);

          if (this.reconnectAttempts < this.maxReconnectAttempts) {
            this.scheduleReconnect();
          }
        };

        this.ws.onerror = (error) => {
          console.error('❌ WebSocket error:', error);
          reject(error);
        };
      } catch (error) {
        console.error('❌ Error creating WebSocket connection:', error);
        reject(error);
      }
    });
  }

  private handleMessage(message: WebSocketMessage) {
    switch (message.type) {
      case 'train_position':
        if (message.data) {
          this.notifyTrainUpdate(message.data);
        }
        break;

      case 'train_positions':
        if (message.data) {
          this.notifyAllTrainsUpdate(message.data);
        }
        break;

      case 'pong':
        console.log('📡 Server ping response received');
        break;

      case 'error':
        console.error('❌ Server error:', message.data);
        break;

      default:
        console.log('❓ Unknown message type:', message.type);
    }
  }

  private scheduleReconnect() {
    this.reconnectAttempts++;
    const delay = this.reconnectDelay * Math.pow(2, this.reconnectAttempts - 1);

    console.log(
      `🔄 Attempting to reconnect in ${delay}ms (attempt ${this.reconnectAttempts}/${this.maxReconnectAttempts})`
    );

    setTimeout(() => {
      if (!this.isConnected) {
        this.connect().catch((error) => {
          console.error('❌ Reconnection failed:', error);
        });
      }
    }, delay);
  }

  disconnect() {
    if (this.ws) {
      this.ws.close(1000, 'Client disconnect');
      this.ws = null;
    }
    this.isConnected = false;
    this.notifyConnectionStatus(false);
  }

  // Subscribe to specific train updates
  subscribeToTrain(trainId: string) {
    if (this.ws && this.isConnected) {
      this.ws.send(
        JSON.stringify({
          type: 'subscribe_train',
          trainId,
        })
      );
    }
  }

  // Unsubscribe from specific train updates
  unsubscribeFromTrain(trainId: string) {
    if (this.ws && this.isConnected) {
      this.ws.send(
        JSON.stringify({
          type: 'unsubscribe_train',
          trainId,
        })
      );
    }
  }

  // Request all train positions
  requestAllTrains() {
    if (this.ws && this.isConnected) {
      this.ws.send(
        JSON.stringify({
          type: 'get_all_trains',
        })
      );
    }
  }

  // Send ping to keep connection alive
  ping() {
    if (this.ws && this.isConnected) {
      this.ws.send(
        JSON.stringify({
          type: 'ping',
          timestamp: Date.now(),
        })
      );
    }
  }

  // Event listeners
  onTrainUpdate(callback: (train: TrainPosition) => void): () => void {
    this.updateCallbacks.push(callback);
    return () => {
      const index = this.updateCallbacks.indexOf(callback);
      if (index > -1) {
        this.updateCallbacks.splice(index, 1);
      }
    };
  }

  onAllTrainsUpdate(callback: (trains: TrainPosition[]) => void): () => void {
    this.allTrainsCallbacks.push(callback);
    return () => {
      const index = this.allTrainsCallbacks.indexOf(callback);
      if (index > -1) {
        this.allTrainsCallbacks.splice(index, 1);
      }
    };
  }

  onConnectionChange(callback: (connected: boolean) => void): () => void {
    this.connectionCallbacks.push(callback);
    return () => {
      const index = this.connectionCallbacks.indexOf(callback);
      if (index > -1) {
        this.connectionCallbacks.splice(index, 1);
      }
    };
  }

  private notifyTrainUpdate(train: TrainPosition) {
    this.updateCallbacks.forEach((callback) => callback(train));
  }

  private notifyAllTrainsUpdate(trains: TrainPosition[]) {
    this.allTrainsCallbacks.forEach((callback) => callback(trains));
  }

  private notifyConnectionStatus(connected: boolean) {
    this.connectionCallbacks.forEach((callback) => callback(connected));
  }

  // Get connection status
  getConnectionStatus(): boolean {
    return this.isConnected;
  }

  // Start automatic ping
  startAutoPing(interval: number = 30000) {
    setInterval(() => {
      this.ping();
    }, interval);
  }
}

// Export singleton instance
export const trainDataService = new TrainDataService();
