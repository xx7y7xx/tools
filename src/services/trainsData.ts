import axios, { AxiosResponse } from 'axios';

// Configuration for the trains-data API
const TRAINS_DATA_BASE_URL = 'http://localhost:3000/trains-data';

// Interface for train data from checiMap.json
export interface TrainInfo {
  from_station: string;
  station_train_code: string;
  to_station: string;
  total_num: string;
  train_no: string;
}

// Interface for the complete trains data response
export interface TrainsDataResponse {
  [trainCode: string]: TrainInfo;
}

/**
 * Fetches trains data from checiMap.json for a specific date
 * @param dateString - Date in YYYYMMDD format (e.g., '20250609'). Required since there's no root checiMap.json
 * @returns Promise containing trains data
 */
export const fetchTrainsData = async (
  dateString: string
): Promise<TrainsDataResponse> => {
  try {
    const response: AxiosResponse = await axios.get(
      `${TRAINS_DATA_BASE_URL}/${dateString}/checiMap.json`,
      {
        headers: {
          Accept: 'application/json',
        },
      }
    );

    return response.data;
  } catch (error) {
    console.error('Error fetching trains data:', error);
    throw new Error(
      `Failed to fetch trains data for ${dateString}: ${
        error instanceof Error ? error.message : 'Unknown error'
      }`
    );
  }
};

/**
 * Fetches data for a specific train across multiple dates
 * @param trainCode - Train code (e.g., '1461')
 * @param dates - Array of dates in YYYYMMDD format
 * @returns Promise containing train data trends
 */
export const fetchTrainTrends = async (
  trainCode: string,
  dates: string[]
): Promise<{ date: string; total_num: number; trainInfo?: TrainInfo }[]> => {
  const results = await Promise.allSettled(
    dates.map(async (date) => {
      try {
        const data = await fetchTrainsData(date);
        const trainInfo = data[trainCode];
        return {
          date,
          total_num: trainInfo ? parseInt(trainInfo.total_num) : 0,
          trainInfo: trainInfo || undefined,
        };
      } catch (error) {
        return {
          date,
          total_num: 0,
          trainInfo: undefined,
        };
      }
    })
  );

  return results
    .filter(
      (result): result is PromiseFulfilledResult<any> =>
        result.status === 'fulfilled'
    )
    .map((result) => result.value)
    .sort((a, b) => a.date.localeCompare(b.date));
};
