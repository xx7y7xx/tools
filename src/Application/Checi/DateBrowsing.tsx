import React, { useState } from 'react';
import {
  Card,
  Space,
  DatePicker,
  Input,
  Typography,
  List,
  Tag,
  Alert,
} from 'antd';
import { SearchOutlined, CalendarOutlined } from '@ant-design/icons';
import { fetchTrainsData, TrainInfo } from '../../services/trainsData';
import dayjs from 'dayjs';

const { Search } = Input;
const { Title, Paragraph } = Typography;

interface DateBrowsingProps {
  onError: (error: string | null) => void;
  onLoadingChange: (loading: boolean) => void;
}

const DateBrowsing: React.FC<DateBrowsingProps> = ({
  onError,
  onLoadingChange,
}) => {
  const [loading, setLoading] = useState(false);
  const [selectedDate, setSelectedDate] = useState<string>('');
  const [trains, setTrains] = useState<{ [key: string]: TrainInfo }>({});
  const [searchTerm, setSearchTerm] = useState('');

  const handleDateSelect = async (date: dayjs.Dayjs | null) => {
    if (!date) {
      setSelectedDate('');
      setTrains({});
      return;
    }

    const dateString = date.format('YYYYMMDD');
    setSelectedDate(dateString);
    setLoading(true);
    onLoadingChange(true);
    onError(null);

    try {
      const data = await fetchTrainsData(dateString);
      setTrains(data || {});
    } catch (err) {
      console.error('Failed to fetch trains data:', err);
      onError(err instanceof Error ? err.message : '获取列车数据失败');
      setTrains({});
    } finally {
      setLoading(false);
      onLoadingChange(false);
    }
  };

  const handleSearch = (value: string) => {
    setSearchTerm(value.trim());
  };

  const filteredTrains = Object.entries(trains).filter(([trainCode, train]) => {
    if (!searchTerm) return true;
    const searchLower = searchTerm.toLowerCase();
    return (
      trainCode.toLowerCase().includes(searchLower) ||
      train.from_station.toLowerCase().includes(searchLower) ||
      train.to_station.toLowerCase().includes(searchLower)
    );
  });

  return (
    <Space direction="vertical" size="large" style={{ width: '100%' }}>
      <div>
        <Title level={4}>按日期浏览列车数据</Title>
        <Paragraph>
          选择日期查看当天的所有列车数据，可以使用搜索框按车次号或站点筛选
        </Paragraph>
        <Space>
          <DatePicker
            placeholder="选择日期"
            onChange={handleDateSelect}
            disabledDate={(date) => date && date > dayjs().endOf('day')}
            style={{ width: 200 }}
          />
          <Search
            placeholder="搜索车次号或站点"
            allowClear
            onSearch={handleSearch}
            style={{ width: 300 }}
            disabled={!selectedDate}
          />
        </Space>
      </div>

      {loading ? (
        <Card loading={true} />
      ) : selectedDate ? (
        filteredTrains.length > 0 ? (
          <Card
            title={
              <Space>
                <CalendarOutlined />
                <span>{selectedDate} 列车数据</span>
                <Tag color="blue">{filteredTrains.length} 个车次</Tag>
              </Space>
            }
          >
            <List
              dataSource={filteredTrains}
              renderItem={([trainCode, train]) => (
                <List.Item
                  key={trainCode}
                  extra={
                    <Space>
                      <Tag color="green">{train.total_num}</Tag>
                      <Tag color="blue">{train.train_no}</Tag>
                    </Space>
                  }
                >
                  <List.Item.Meta
                    title={trainCode}
                    description={
                      <Space>
                        <span>{train.from_station}</span>
                        <span>→</span>
                        <span>{train.to_station}</span>
                      </Space>
                    }
                  />
                </List.Item>
              )}
            />
          </Card>
        ) : (
          <Alert
            message="未找到数据"
            description={
              searchTerm
                ? '没有找到匹配的列车数据，请尝试其他搜索条件'
                : '该日期没有列车数据'
            }
            type="info"
            showIcon
          />
        )
      ) : (
        <Alert
          message="请选择日期"
          description="请先选择一个日期来查看列车数据"
          type="info"
          showIcon
        />
      )}
    </Space>
  );
};

export default DateBrowsing;
