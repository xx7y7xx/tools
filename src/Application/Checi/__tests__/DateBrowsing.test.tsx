import React from 'react';
import { render, screen, fireEvent, waitFor } from '@testing-library/react';
import '@testing-library/jest-dom';
import DateBrowsing from '../DateBrowsing';
import { fetchTrainsData } from '../../../services/trainsData';
import { TrainInfo } from '../../../services/trainsData';
import * as antd from 'antd';

// Mock Ant Design's responsive observer to prevent addListener error
jest.mock('antd/lib/_util/responsiveObserver', () => ({
  __esModule: true,
  default: () => ({
    subscribe: jest.fn(),
    unsubscribe: jest.fn(),
  }),
}));

// Mock Ant Design components
jest.mock('antd', () => {
  const antd = jest.requireActual('antd');
  return {
    ...antd,
    List: Object.assign(
      ({
        dataSource,
        renderItem,
      }: {
        dataSource: TrainInfo[];
        renderItem: (item: TrainInfo) => React.ReactNode;
      }) => (
        <div data-testid="mock-list">
          {dataSource?.map((item: TrainInfo, index: number) => (
            <div key={index} data-testid="mock-list-item">
              {renderItem(item)}
            </div>
          ))}
        </div>
      ),
      {
        Item: ({ children }: { children: React.ReactNode }) => (
          <div data-testid="mock-list-item-component">{children}</div>
        ),
      }
    ),
  };
});

jest.mock('../../../services/trainsData', () => ({
  fetchTrainsData: jest.fn(),
}));

// Replace the class-based MockInputSearch with a function component using hooks and forwardRef
const MockInputSearch = React.forwardRef<any, any>(
  (
    {
      placeholder,
      style,
      enterButton,
      onSearch,
      onChange,
      value = '',
      ...rest
    },
    ref
  ) => {
    return (
      <div>
        <input
          ref={ref}
          placeholder={placeholder}
          style={style}
          data-testid={placeholder}
          value={value}
          onChange={onChange}
          {...rest}
        />
        <button onClick={() => onSearch && onSearch(value)}>
          {enterButton}
        </button>
      </div>
    );
  }
);
antd.Input.Search = MockInputSearch;

describe('DateBrowsing', () => {
  const mockTrainsData = {
    G1: {
      train_no: 'T100',
      from_station: '北京',
      to_station: '上海',
      station_train_code: 'G1',
      total_num: '1',
    },
    G2: {
      train_no: 'T200',
      from_station: '上海',
      to_station: '广州',
      station_train_code: 'G2',
      total_num: '1',
    },
  };

  beforeEach(() => {
    jest.clearAllMocks();
    (fetchTrainsData as jest.Mock).mockResolvedValue(mockTrainsData);
  });

  it('should handle happy path - fetch and display data', async () => {
    const onError = jest.fn();
    const onLoadingChange = jest.fn();
    render(
      <DateBrowsing onError={onError} onLoadingChange={onLoadingChange} />
    );

    // Initial state
    expect(screen.getByText('暂无数据')).toBeInTheDocument();
    expect(onLoadingChange).not.toHaveBeenCalled();
    expect(onError).not.toHaveBeenCalled();

    // Enter date and fetch data
    const dateInput = screen.getByPlaceholderText('输入日期 (如: 20250609)');
    const searchButton = screen.getByText('下载数据');
    fireEvent.change(dateInput, { target: { value: '20241120' } });
    fireEvent.click(searchButton);

    // Check loading state
    expect(onLoadingChange).toHaveBeenCalledWith(true);
    expect(fetchTrainsData).toHaveBeenCalledWith('20241120');

    // Wait for data to load
    await waitFor(() => {
      expect(
        screen.getByText('已加载 20241120 的数据 (2 趟列车)')
      ).toBeInTheDocument();
    });

    // Check final state
    expect(onLoadingChange).toHaveBeenCalledWith(false);
    expect(onError).toHaveBeenCalledWith(null); // Component calls onError(null) to clear errors
    expect(screen.getByText('🚄 G1')).toBeInTheDocument();
    expect(screen.getByText('🚄 G2')).toBeInTheDocument();
    expect(screen.getByText('北京')).toBeInTheDocument();
    expect(screen.getAllByText('上海').length).toBe(2);
    expect(screen.getByText('广州')).toBeInTheDocument();
  });

  it('matches snapshot', async () => {
    const onError = jest.fn();
    const onLoadingChange = jest.fn();
    const { container } = render(
      <DateBrowsing onError={onError} onLoadingChange={onLoadingChange} />
    );

    // Enter date and fetch data
    const dateInput = screen.getByPlaceholderText('输入日期 (如: 20250609)');
    const searchButton = screen.getByText('下载数据');
    fireEvent.change(dateInput, { target: { value: '20241120' } });
    fireEvent.click(searchButton);

    // Wait for data to load
    await waitFor(() => {
      expect(
        screen.getByText('已加载 20241120 的数据 (2 趟列车)')
      ).toBeInTheDocument();
    });

    // Create snapshot
    expect(container).toMatchSnapshot();
  });
});
