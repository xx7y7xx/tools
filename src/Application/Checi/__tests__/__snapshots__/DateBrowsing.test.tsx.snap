// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DateBrowsing (snapshot) matches snapshot 1`] = `
<div>
  <div
    class="ant-space css-dev-only-do-not-override-1uq9j6g ant-space-vertical ant-space-gap-row-large ant-space-gap-col-large"
    style="width: 100%;"
  >
    <div
      class="ant-space-item"
    >
      <div>
        <h4
          class="ant-typography css-dev-only-do-not-override-1uq9j6g"
        >
          <span
            aria-label="download"
            class="anticon anticon-download"
            role="img"
          >
            <svg
              aria-hidden="true"
              data-icon="download"
              fill="currentColor"
              focusable="false"
              height="1em"
              viewBox="64 64 896 896"
              width="1em"
            >
              <path
                d="M505.7 661a8 8 0 0012.6 0l112-141.7c4.1-5.2.4-12.9-6.3-12.9h-74.1V168c0-4.4-3.6-8-8-8h-60c-4.4 0-8 3.6-8 8v338.3H400c-6.7 0-10.4 7.7-6.3 12.9l112 141.8zM878 626h-60c-4.4 0-8 3.6-8 8v154H214V634c0-4.4-3.6-8-8-8h-60c-4.4 0-8 3.6-8 8v198c0 17.7 14.3 32 32 32h684c17.7 0 32-14.3 32-32V634c0-4.4-3.6-8-8-8z"
              />
            </svg>
          </span>
           选择日期并加载数据
        </h4>
        <div
          class="ant-typography css-dev-only-do-not-override-1uq9j6g"
        >
          输入日期一次性下载该日期的所有列车数据到本地 (格式: YYYYMMDD)
          <br />
          <strong>
            可用日期示例:
          </strong>
           20250609, 20250608, 20250607, 20250302, 20241120
        </div>
        <div
          class="ant-space css-dev-only-do-not-override-1uq9j6g ant-space-horizontal ant-space-align-center ant-space-gap-row-small ant-space-gap-col-small"
        >
          <div
            class="ant-space-item"
          >
            <span
              class="ant-input-group-wrapper ant-input-group-wrapper-lg ant-input-group-wrapper-outlined css-dev-only-do-not-override-1uq9j6g ant-input-search ant-input-search-large ant-input-search-with-button"
              style="width: 300px;"
            >
              <span
                class="ant-input-wrapper ant-input-group css-dev-only-do-not-override-1uq9j6g"
              >
                <span
                  class="ant-input-affix-wrapper ant-input-affix-wrapper-lg css-dev-only-do-not-override-1uq9j6g ant-input-outlined"
                >
                  <input
                    class="ant-input ant-input-lg css-dev-only-do-not-override-1uq9j6g"
                    placeholder="输入日期 (如: 20250609)"
                    type="text"
                    value="20241120"
                  />
                  <span
                    class="ant-input-suffix"
                  >
                    <span
                      class="ant-input-clear-icon"
                      role="button"
                      tabindex="-1"
                    >
                      <span
                        aria-label="close-circle"
                        class="anticon anticon-close-circle"
                        role="img"
                      >
                        <svg
                          aria-hidden="true"
                          data-icon="close-circle"
                          fill="currentColor"
                          fill-rule="evenodd"
                          focusable="false"
                          height="1em"
                          viewBox="64 64 896 896"
                          width="1em"
                        >
                          <path
                            d="M512 64c247.4 0 448 200.6 448 448S759.4 960 512 960 64 759.4 64 512 264.6 64 512 64zm127.98 274.82h-.04l-.08.06L512 466.75 384.14 338.88c-.04-.05-.06-.06-.08-.06a.12.12 0 00-.07 0c-.03 0-.05.01-.09.05l-45.02 45.02a.2.2 0 00-.05.09.12.12 0 000 .07v.02a.27.27 0 00.06.06L466.75 512 338.88 639.86c-.05.04-.06.06-.06.08a.12.12 0 000 .07c0 .03.01.05.05.09l45.02 45.02a.2.2 0 00.09.05.12.12 0 00.07 0c.02 0 .04-.01.08-.05L512 557.25l127.86 127.87c.04.04.06.05.08.05a.12.12 0 00.07 0c.03 0 .05-.01.09-.05l45.02-45.02a.2.2 0 00.05-.09.12.12 0 000-.07v-.02a.27.27 0 00-.05-.06L557.25 512l127.87-127.86c.04-.04.05-.06.05-.08a.12.12 0 000-.07c0-.03-.01-.05-.05-.09l-45.02-45.02a.2.2 0 00-.09-.05.12.12 0 00-.07 0z"
                          />
                        </svg>
                      </span>
                    </span>
                  </span>
                </span>
                <span
                  class="ant-input-group-addon"
                >
                  <button
                    class="ant-btn css-dev-only-do-not-override-1uq9j6g ant-btn-primary ant-btn-lg ant-input-search-button"
                    type="button"
                  >
                    <span>
                      下载数据
                    </span>
                  </button>
                </span>
              </span>
            </span>
          </div>
          <div
            class="ant-space-item"
          >
            <span
              class="ant-tag ant-tag-success css-dev-only-do-not-override-1uq9j6g"
              style="font-size: 14px; padding: 4px 8px;"
            >
              已加载 
              20241120
               的数据 (
              2
               趟列车)
            </span>
          </div>
        </div>
      </div>
    </div>
    <div
      class="ant-space-item"
    >
      <div>
        <h4
          class="ant-typography css-dev-only-do-not-override-1uq9j6g"
        >
          <span
            aria-label="filter"
            class="anticon anticon-filter"
            role="img"
          >
            <svg
              aria-hidden="true"
              data-icon="filter"
              fill="currentColor"
              focusable="false"
              height="1em"
              viewBox="64 64 896 896"
              width="1em"
            >
              <path
                d="M880.1 154H143.9c-24.5 0-39.8 26.7-27.5 48L349 597.4V838c0 17.7 14.2 32 31.8 32h262.4c17.6 0 31.8-14.3 31.8-32V597.4L907.7 202c12.2-21.3-3.1-48-27.6-48zM603.4 798H420.6V642h182.9v156zm9.6-236.6l-9.5 16.6h-183l-9.5-16.6L212.7 226h598.6L613 561.4z"
              />
            </svg>
          </span>
           按车站筛选 (本地搜索，无需重新下载)
        </h4>
        <div
          class="ant-typography css-dev-only-do-not-override-1uq9j6g"
        >
          从已下载的 
          <span
            class="ant-tag ant-tag-blue css-dev-only-do-not-override-1uq9j6g"
          >
            20241120
          </span>
           
          数据中搜索包含指定车站的列车
        </div>
        <div
          class="ant-space css-dev-only-do-not-override-1uq9j6g ant-space-horizontal ant-space-align-center ant-space-gap-row-small ant-space-gap-col-small"
        >
          <div
            class="ant-space-item"
          >
            <span
              class="ant-input-group-wrapper ant-input-group-wrapper-lg ant-input-group-wrapper-outlined css-dev-only-do-not-override-1uq9j6g ant-input-search ant-input-search-large ant-input-search-with-button"
              style="width: 300px;"
            >
              <span
                class="ant-input-wrapper ant-input-group css-dev-only-do-not-override-1uq9j6g"
              >
                <span
                  class="ant-input-affix-wrapper ant-input-affix-wrapper-lg css-dev-only-do-not-override-1uq9j6g ant-input-outlined"
                >
                  <input
                    class="ant-input ant-input-lg css-dev-only-do-not-override-1uq9j6g"
                    placeholder="输入车站名称 (如: 北京, 上海)"
                    type="text"
                    value=""
                  />
                  <span
                    class="ant-input-suffix"
                  >
                    <span
                      class="ant-input-clear-icon ant-input-clear-icon-hidden"
                      role="button"
                      tabindex="-1"
                    >
                      <span
                        aria-label="close-circle"
                        class="anticon anticon-close-circle"
                        role="img"
                      >
                        <svg
                          aria-hidden="true"
                          data-icon="close-circle"
                          fill="currentColor"
                          fill-rule="evenodd"
                          focusable="false"
                          height="1em"
                          viewBox="64 64 896 896"
                          width="1em"
                        >
                          <path
                            d="M512 64c247.4 0 448 200.6 448 448S759.4 960 512 960 64 759.4 64 512 264.6 64 512 64zm127.98 274.82h-.04l-.08.06L512 466.75 384.14 338.88c-.04-.05-.06-.06-.08-.06a.12.12 0 00-.07 0c-.03 0-.05.01-.09.05l-45.02 45.02a.2.2 0 00-.05.09.12.12 0 000 .07v.02a.27.27 0 00.06.06L466.75 512 338.88 639.86c-.05.04-.06.06-.06.08a.12.12 0 000 .07c0 .03.01.05.05.09l45.02 45.02a.2.2 0 00.09.05.12.12 0 00.07 0c.02 0 .04-.01.08-.05L512 557.25l127.86 127.87c.04.04.06.05.08.05a.12.12 0 00.07 0c.03 0 .05-.01.09-.05l45.02-45.02a.2.2 0 00.05-.09.12.12 0 000-.07v-.02a.27.27 0 00-.05-.06L557.25 512l127.87-127.86c.04-.04.05-.06.05-.08a.12.12 0 000-.07c0-.03-.01-.05-.05-.09l-45.02-45.02a.2.2 0 00-.09-.05.12.12 0 00-.07 0z"
                          />
                        </svg>
                      </span>
                    </span>
                  </span>
                </span>
                <span
                  class="ant-input-group-addon"
                >
                  <button
                    class="ant-btn css-dev-only-do-not-override-1uq9j6g ant-btn-primary ant-btn-lg ant-input-search-button"
                    type="button"
                  >
                    <span>
                      筛 选
                    </span>
                  </button>
                </span>
              </span>
            </span>
          </div>
          <div
            class="ant-space-item"
          />
        </div>
      </div>
    </div>
    <div
      class="ant-space-item"
    >
      <div
        class="ant-card ant-card-bordered css-dev-only-do-not-override-1uq9j6g"
        style="margin-top: 20px;"
      >
        <div
          class="ant-card-head"
        >
          <div
            class="ant-card-head-wrapper"
          >
            <div
              class="ant-card-head-title"
            >
              <span>
                🚄 列车信息
                <span>
                   (共 
                  2
                   趟列车)
                </span>
              </span>
            </div>
          </div>
        </div>
        <div
          class="ant-card-body"
        >
          <div
            data-testid="mock-list"
          >
            <div
              data-testid="mock-list-item"
            >
              <div
                data-testid="mock-list-item-component"
              >
                <div
                  class="ant-card ant-card-bordered ant-card-hoverable ant-card-small css-dev-only-do-not-override-1uq9j6g"
                  style="width: 100%;"
                >
                  <div
                    class="ant-card-body"
                  >
                    <div
                      style="display: flex; justify-content: space-between; align-items: center;"
                    >
                      <div
                        style="flex: 1;"
                      >
                        <h5
                          class="ant-typography css-dev-only-do-not-override-1uq9j6g"
                          style="margin: 0px 0px 8px 0px;"
                        >
                          🚄 
                          G1
                        </h5>
                        <div
                          class="ant-space css-dev-only-do-not-override-1uq9j6g ant-space-vertical ant-space-gap-row-small ant-space-gap-col-small"
                        >
                          <div
                            class="ant-space-item"
                          >
                            <div>
                              <strong>
                                起点站:
                              </strong>
                               
                              北京
                            </div>
                          </div>
                          <div
                            class="ant-space-item"
                          >
                            <div>
                              <strong>
                                终点站:
                              </strong>
                               
                              上海
                            </div>
                          </div>
                          <div
                            class="ant-space-item"
                          >
                            <div>
                              <strong>
                                车次编号:
                              </strong>
                               
                              T100
                            </div>
                          </div>
                        </div>
                      </div>
                      <div
                        style="text-align: right; color: rgb(102, 102, 102);"
                      >
                        <div>
                          <strong>
                            总数:
                          </strong>
                        </div>
                        <div
                          style="font-size: 24px; font-weight: bold;"
                        >
                          1
                        </div>
                      </div>
                    </div>
                  </div>
                </div>
              </div>
            </div>
            <div
              data-testid="mock-list-item"
            >
              <div
                data-testid="mock-list-item-component"
              >
                <div
                  class="ant-card ant-card-bordered ant-card-hoverable ant-card-small css-dev-only-do-not-override-1uq9j6g"
                  style="width: 100%;"
                >
                  <div
                    class="ant-card-body"
                  >
                    <div
                      style="display: flex; justify-content: space-between; align-items: center;"
                    >
                      <div
                        style="flex: 1;"
                      >
                        <h5
                          class="ant-typography css-dev-only-do-not-override-1uq9j6g"
                          style="margin: 0px 0px 8px 0px;"
                        >
                          🚄 
                          G2
                        </h5>
                        <div
                          class="ant-space css-dev-only-do-not-override-1uq9j6g ant-space-vertical ant-space-gap-row-small ant-space-gap-col-small"
                        >
                          <div
                            class="ant-space-item"
                          >
                            <div>
                              <strong>
                                起点站:
                              </strong>
                               
                              上海
                            </div>
                          </div>
                          <div
                            class="ant-space-item"
                          >
                            <div>
                              <strong>
                                终点站:
                              </strong>
                               
                              广州
                            </div>
                          </div>
                          <div
                            class="ant-space-item"
                          >
                            <div>
                              <strong>
                                车次编号:
                              </strong>
                               
                              T200
                            </div>
                          </div>
                        </div>
                      </div>
                      <div
                        style="text-align: right; color: rgb(102, 102, 102);"
                      >
                        <div>
                          <strong>
                            总数:
                          </strong>
                        </div>
                        <div
                          style="font-size: 24px; font-weight: bold;"
                        >
                          1
                        </div>
                      </div>
                    </div>
                  </div>
                </div>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
